config/v1alpha1/techpreview.imagepolicy.testsuite.yaml

```diff
@@ -0,0 +1,417 @@
+    expectedError: "spec.policy.signedIdentity: Invalid value: \"object\": exactRepository is required when matchPolicy is ExactRepository, and forbidden otherwise"
+  - name: Should not allow signedIdentity matchPolicy RemapIdentity but not set prefixes
+    initial: |
+      apiVersion: config.openshift.io/v1alpha1
+      kind: ImagePolicy
+      spec:
+        scopes:
+        - example.com
+        policy:
+          rootOfTrust:
+            policyType: PublicKey
+            publicKey:
+              keyData: Zm9vIGJhcg==
+          signedIdentity:
+            matchPolicy: RemapIdentity
+    expectedError: "spec.policy.signedIdentity: Invalid value: \"object\": remapIdentity is required when matchPolicy is RemapIdentity, and forbidden otherwise"
+  - name: Test scope should not allow 'busybox'
+    initial: |
+      apiVersion: config.openshift.io/v1alpha1
+      kind: ImagePolicy
+      spec:
+        scopes:
+        - busybox
+        policy:
+          rootOfTrust:
+            policyType: PublicKey
+            publicKey:
+              keyData: Zm9vIGJhcg==
+    expectedError: "spec.scopes[0]: Invalid value: \"string\": invalid image scope format, scope must contain a fully qualified domain name or 'localhost'"
+  - name: Test scope should not allow start with subnamesapces '*.example.com/test'
+    initial: |
+      apiVersion: config.openshift.io/v1alpha1
+      kind: ImagePolicy
+      spec:
+        scopes:
+        - "*.example.com/test"
+        policy:
+          rootOfTrust:
+            policyType: PublicKey
+            publicKey:
+              keyData: Zm9vIGJhcg==
+    expectedError: "spec.scopes[0]: Invalid value: \"string\": invalid image scope with wildcard, a wildcard can only be at the start of the domain and does not contain subdomains"
+  - name: Test scope should not allow invalid digest
+    initial: |
+      apiVersion: config.openshift.io/v1alpha1
+      kind: ImagePolicy
+      spec:
+        scopes:
+        - example.com/namespace/namespace@sha256:12dsdf
+        policy:
+          rootOfTrust:
+            policyType: PublicKey
+            publicKey:
+              keyData: Zm9vIGJhcg==
+    expectedError: "spec.scopes[0]: Invalid value: \"string\": invalid repository namespace or image specification in the image scope"
+  - name: Test should not allow tag in ExactRepository repository
+    initial: |
+      apiVersion: config.openshift.io/v1alpha1
+      kind: ImagePolicy
+      spec:
+        scopes:
+        - example.com
+        policy:
+          rootOfTrust:
+            policyType: PublicKey
+            publicKey:
+              keyData: Zm9vIGJhcg==
+          signedIdentity:
+            matchPolicy: ExactRepository
+            exactRepository: 
+              repository: example.com/namespace/namespace:latest
+    expectedError: "spec.policy.signedIdentity.exactRepository.repository: Invalid value: \"string\": invalid repository in the signedIdentity"
```

Comment:
This test case is possibly obvious, but, maybe we should have a CEL validation that checks that there's no `:` included and then you could add an error saying that the repository should not include an image tag or digest to be more specific

If you think this is something users may do it might be useful

Have you considered making a repository string alias to use in the various place that it's used in the API? I think the validation is the same on repository here and prefix and signedPrefix in remap right?

Reply:
tests added

updated to use alias. The validation are the same